morf-core/src/main/java/org/alfasoftware/morf/jdbc/ResultSetComparer.java

```diff
@@ -641,4 +702,75 @@ public interface CompareCallback {
      */
     void mismatch(ResultSetMismatch mismatch);
   }
+
+
+  /**
+   * Enum class representing validations of the left and right {@link ResultSet}s.
+   *
+   * @author Copyright (c) Alfa Financial Software Limited. 2024
+   */
+  public enum ResultSetValidator {
+
+     NO_VALIDATION((leftRs, rightRs) -> {}),
+
+     NON_ZERO_RECORD_COUNT_ON_LEFT((leftRs, rightRs) -> {
+       try {
+         if (!leftRs.isBeforeFirst()) {
+           throw new IllegalStateException(format("The following query should return at least one record: [%s]", leftRs.getStatement()));
+         }
+      } catch (SQLException e) {
+        ExceptionUtils.rethrow(e);
+      }
+     }),
+
+     NON_ZERO_RECORD_COUNT_ON_RIGHT((leftRs, rightRs) -> {
+       try {
+         if (!rightRs.isBeforeFirst()) {
+           throw new IllegalStateException(format("The following query should return at least one record: [%s]", rightRs.getStatement()));
+         }
+      } catch (SQLException e) {
+        ExceptionUtils.rethrow(e);
+      }
+     }),
+
+     NON_ZERO_RECORD_COUNT_ON_LEFT_AND_RIGHT((leftRs, rightRs) -> {
```

Comment:
Can we remove this one. It should be equivalent to using both the left and right counterparts.

In fact, since you've already the test case set up it should be easy to prove and increase test coverage all at the same time.